cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1642,42 +1576,20 @@ public void onFailure(int statusCode, String response, Throwable throwable) {
     public void setSubscriptionAttribute(String attributeId, String value) {
         this.waitForTrackingConsent(() -> new Thread(() -> this.getSubscriptionId(subscriptionId -> {
             if (subscriptionId != null) {
-                JSONObject jsonBody = new JSONObject();
+                JSONObject jsonBody = getJsonObject();
                 try {
-                    jsonBody.put("channelId", this.channelId);
+                    jsonBody.put("channelId", getChannelId(getContext()));
```

Comment:
we will need it to mock the channel id from test cases and it will be the same as this.channelId because we are saving channel id to local preferences every time we do init with a different channel id